app/lib/features/tasks/providers/notifiers.dart

```diff
@@ -86,3 +90,41 @@ class TaskNotifier extends FamilyAsyncNotifier<Task, Task> {
     return task;
   }
 }
+
+//List of all task list
+//If space Id given then it will return list of space task list
+class AsyncTaskListsNotifier
+    extends FamilyAsyncNotifier<List<TaskList>, String?> {
+  late Stream<bool> _listener;
+  late StreamSubscription<bool> _poller;
+
+  @override
+  Future<List<TaskList>> build(String? arg) async {
+    final client = ref.watch(alwaysClientProvider);
+
+    //GET ALL TASKS LIST
+    _listener = client.subscribeStream('tasks');
+
+    _poller = _listener.listen((e) async {
+      state = await AsyncValue.guard(() => _getTasksList(client));
+    });
+    ref.onDispose(() => _poller.cancel());
+
+    return await _getTasksList(client);
+  }
+
+  Future<List<TaskList>> _getTasksList(Client client) async {
+    //GET ALL TASKS LIST
+    final tasksList = (await client.taskLists()).toList();
+    // this might throw internally
+
+    if (arg == null) {
+      return tasksList;
+    } else {
+      //GET SPACE TASKS LIST
+      return tasksList
+          .where((t) => t.spaceIdStr() == arg!)
+          .toList(); // this might throw internally
```

Comment:
why would that throw internally?
```suggestion
          .toList();
```

Reply:
I copied code from pins notifiers and comment was there. I don't know if it is applicable to tasks module or not but this comment is also there in events notifiers.